src/sentry/services/hybrid_cloud/__init__.py

```diff
@@ -121,21 +124,27 @@ def get_organizations(
 
     @abstractmethod
     def get_organization_by_slug(
-        self, slug: str, only_visible: bool, allow_stale: bool
+        self, user_id: Optional[int], slug: str, only_visible: bool, allow_stale: bool
```

Comment:
The `get_organization_by_slug` is often used by super user call sites that really do want an organization even if the current user does not have a relationship directly to it.  This `user_id` optionally would fetch the membership, if it exists, into the `ApiOrganization` object, see above for my thinking.

Reply:
Should you use a `*` separator to force keyword argument usage?

Reply:
Good point, I can change that.